src/edit_mode/vi/command.rs

```diff
@@ -150,10 +176,44 @@ impl Command {
             // Whenever a motion is required to finish the command we must be in visual mode
             Self::Delete | Self::Change => vec![ReedlineOption::Edit(EditCommand::CutSelection)],
             Self::Incomplete => vec![ReedlineOption::Incomplete],
-            Command::RepeatLastAction => match &vi_state.previous {
+            Self::RepeatLastAction => match &vi_state.previous {
                 Some(event) => vec![ReedlineOption::Event(event.clone())],
                 None => vec![],
             },
+            Self::ChangeInside(left) if is_valid_change_inside_left(left) => {
+                let right = bracket_for(left);
+                vec![
+                    ReedlineOption::Edit(EditCommand::CutLeftBefore(*left)),
+                    ReedlineOption::Edit(EditCommand::CutRightBefore(right)),
+                ]
+            }
+            Self::ChangeInside(right) if is_valid_change_inside_right(right) => {
+                let left = bracket_for(right);
+                vec![
+                    ReedlineOption::Edit(EditCommand::CutLeftBefore(left)),
+                    ReedlineOption::Edit(EditCommand::CutRightBefore(*right)),
+                ]
+            }
```

Comment:
I am thinking you could save yourself some duplication here by doing the conversion to the one bracket side when creating the `Change/DeleteInisde(char)` but then there would be also something to watch out for correctness.

Reply:
I would still need to know if it is a right of left side bracket to get the ordering right.. It felt cleaner to have be a separate match statements. I did collapse the Delete and Change conditions together since they were inherently the same.